packages/test/test-version-utils/src/test/versionUtils.spec.ts

```diff
@@ -3,134 +3,115 @@
  * Licensed under the MIT License.
  */
 import { strict as assert } from "assert";
-import { getRequestedRange, versionHasMovedSparsedMatrix } from "../versionUtils.js";
+import { satisfies } from "semver";
+import { getRequestedVersion, versionHasMovedSparsedMatrix } from "../versionUtils.js";
 
-describe("versionUtils", () => {
-	it("Get the major version number above or below the baseVersion", () => {
-		// assert for major bumps
-		assert.strictEqual(getRequestedRange("1.0.0", -1), "^0.59.0-0");
-		assert.strictEqual(getRequestedRange("1.0.0", -2), "^0.58.0-0");
-		assert.strictEqual(getRequestedRange("1.0.0", 1), "^2.0.0-0");
-		assert.strictEqual(getRequestedRange("2.0.0", -1), "^1.0.0-0");
-
-		// assert for internal release
-		assert.strictEqual(getRequestedRange("2.0.0-internal.1.0.0", -1), "^1.0.0-0");
-		assert.strictEqual(getRequestedRange("2.0.0-internal.1.1.0", -1), "^1.0.0-0");
-		assert.strictEqual(getRequestedRange("2.0.0-internal.1.1.1", -1), "^1.0.0-0");
-		assert.strictEqual(getRequestedRange("2.0.0-internal.2.0.0", -2), "^1.0.0-0");
+const checkRequestedVersionSatisfies = (baseVersion, requested, expectedVersion) => {
+	const version = getRequestedVersion(baseVersion, requested);
+	assert(
+		satisfies(version, expectedVersion),
+		`getRequestedVersion("${baseVersion}", ${requested}) -> ${version} does not satisfy ${expectedVersion}`,
+	);
+};
 
-		assert.strictEqual(
-			getRequestedRange("2.0.0-internal.2.0.0", -1),
-			">=2.0.0-internal.1.0.0 <2.0.0-internal.2.0.0",
-		);
-		assert.strictEqual(
-			getRequestedRange("2.0.0-internal.2.1.1", -1),
-			">=2.0.0-internal.1.0.0 <2.0.0-internal.2.0.0",
-		);
-		assert.strictEqual(getRequestedRange("2.0.0-internal.2.0.0", -3), "^0.58.0-0");
-		assert.strictEqual(getRequestedRange("2.0.0-internal.2.0.1", -2), "^1.0.0-0");
-		assert.strictEqual(getRequestedRange("2.0.0-internal.1.4.2", -1), "^1.0.0-0");
-		assert.strictEqual(getRequestedRange("2.0.0-internal.1.4.2", -2), "^0.59.0-0");
+describe("versionUtils", () => {
+	describe("getRequestedVersion", () => {
+		it("bumping down public releases", () => {
+			checkRequestedVersionSatisfies("1.0.0", -1, "^0.59.0");
+			checkRequestedVersionSatisfies("1.0.0", -2, "^0.58.0");
+			checkRequestedVersionSatisfies("2.0.0", -1, "^1.0.0");
+			checkRequestedVersionSatisfies("2.3.5", -1, "^1.0.0");
+		});
 
-		assert.strictEqual(getRequestedRange("2.0.0-internal.1.2.3", -1), "^1.0.0-0");
-		assert.strictEqual(
-			getRequestedRange("2.0.0-internal.2.1.0", -1),
-			">=2.0.0-internal.1.0.0 <2.0.0-internal.2.0.0",
-		);
-		assert.strictEqual(
-			getRequestedRange("2.0.0-internal.3.0.0", -1),
-			">=2.0.0-internal.2.0.0 <2.0.0-internal.3.0.0",
-		);
-		assert.strictEqual(
-			getRequestedRange("2.0.0-internal.3.0.0", -2),
-			">=2.0.0-internal.1.0.0 <2.0.0-internal.2.0.0",
-		);
+		it("bumping up public releases", () => {
+			checkRequestedVersionSatisfies("0.59.0", 1, "^1.0.0");
+			checkRequestedVersionSatisfies("0.58.0", 1, "^1.0.0");
+		});
```

Comment:
Do we ever call getRequestedVersion with a positive value outside this test code? If not, I'm wondering if we should put a type constraint on the value to only permit negative integers. If we do need to adjust versions up, then we should consider re-using functions from version-tools.

Reply:
I don't believe that we do. I agree it's probably best to just constraint this behavior for now